nuclear/cmake/Modules/NUClearCompilerSettings.cmake

```diff
@@ -24,3 +24,6 @@ if(NOT MSVC)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 endif()
+
+# Set OpenCL Target version
+add_compile_definitions(CL_TARGET_OPENCL_VERSION=120)
```

Comment:
This shouldn't go here

Reply:
Where should it go?

Reply:
I'm not sure, but this is setting the property for all files in the entire codebase. It's something that should be fixed by the visual mesh

Reply:
Did having this in the ImageCompressor and VisualMesh modules not resolve this warning?

Reply:
I changed the flag inside the module/VisualMesh and it was still giving the error from  nuclear/roles/CMakeFiles/balllocalisation.dir/balllocalisation.cpp.o and similar others from nuclear/roles/CMakeFiles/...

Reply:
It's something that needs to be fixed by the VisualMesh rather than this code

Reply:
Yea, I think you need to define it in the header file itself, not as a command line define as it spreads to things that include it